rest/model/host.go

```diff
@@ -114,14 +115,16 @@ func (apiHost *APIHost) buildFromHostStruct(h interface{}) error {
 	apiHost.DisplayName = ToStringPtr(v.DisplayName)
 	apiHost.HomeVolumeID = ToStringPtr(v.HomeVolumeID)
 
-	imageId, err := v.Distro.GetImageID()
-	if err != nil {
-		return errors.Wrap(err, "problem getting image ID")
-	}
 	di := DistroInfo{
 		Id:       ToStringPtr(v.Distro.Id),
 		Provider: ToStringPtr(v.Distro.Provider),
-		ImageId:  ToStringPtr(imageId),
+	}
+	if !graphQL {
```

Comment:
It's fine to change the signature of this function but I'm not seeing why this branch is needed. Are you mostly trying to avoid doing an extra query for data you won't use?

Reply:
I want to avoid the extra query, and I was also getting an error from that query.

Reply:
What was the error? I think the extra query is fine and not really worth making the code more complex

Reply:
That's mostly a matter of the data used to test being valid. If you look at the function returning the last error, you should be able to find the right field to set for the provider settings